// Imported from
// //depot/google3/google/internal/chrome/cros/edu/schooltools/v1/roster.proto

syntax = "proto3";

package boca;

option optimize_for = LITE_RUNTIME;

message Roster {
  string roster_id = 1;

  string title = 2;

  repeated StudentGroup student_groups = 3;
}

message StudentGroup {
  string title = 1;

  repeated UserIdentity students = 2;

  enum GroupSource {
    GROUP_SOURCE_UNKNOWN = 0;
    CLASSROOM = 1;
    JOIN_CODE = 2;
  }

  GroupSource group_source = 3;

  string student_group_id = 4;
}

message UserIdentity {
  string gaia_id = 1;
  string email = 2;
  string full_name = 3;
  string photo_url = 4;
}
